### PR TITLE
CalendarHasNoActiveDays rule implemented

### PR DESCRIPTION
**Pull request by aannie-y**
Rule to check that the calendar has at least one active (available) day. Used a simple for loop and if statement to implement the rule. Loops through all the calendars in the given ArrayList of GtfsCalendar and checks whether the sum of the enum values from monday->sunday is less than 1. If it is less than 1, then all days are unavailable and a notice should be called.

### CODE REVIEW

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    // Loop through calendar and add up enum values for each day. If value is less than 1, then no
+    // days are active.
+    for (GtfsCalendar calendar : calendarTable.getEntities()) {
+      if ((calendar.monday().getNumber() + calendar.tuesday().getNumber()
+              + calendar.wednesday().getNumber() + calendar.thursday().getNumber()
+              + calendar.friday().getNumber() + calendar.saturday().getNumber()
+              + calendar.sunday().getNumber())
+          < 1) {
```
Comment (jupi3200) [inline gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: Is there any guarantee that getNumber is positive for all enums?
Could there be an overflow?

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      GtfsCalendarService friday, GtfsCalendarService saturday, GtfsCalendarService sunday) {
+    super(new ImmutableMap.Builder<String, Object>()
+              .put("serviceId", serviceId)
+              .put("monday", monday)
+              .put("tuesday", tuesday)
+              .put("wednesday", wednesday)
+              .put("thursday", thursday)
+              .put("friday", friday)
+              .put("saturday", saturday)
+              .put("sunday", sunday)
```
Comment (KathleenQ) [inline gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java]: I'm considering if you already know there must be no active day to generate the notice, you thus directly know all Monday to Sunday are NOT_AVAILABLE and do not need to put their 7 values into the notice. 
Instead, you can put some other fields in the notice, like csvRowNumber of the corresponding calendar, which is very useful.

Comment (jupi3200) [inline gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java]: Wouldn't all possible CalendarHasNoActiveDaysNotice have the same value for monday, tuesday, wednesday etc.?

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendar;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarService;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarTableContainer;
+
+@RunWith(JUnit4.class)
+public class CalendarHasNoActiveDaysValidatorTest {
+  private CalendarHasNoActiveDaysValidator validator = new CalendarHasNoActiveDaysValidator();
+
+  private final GtfsCalendar all_active_calendar =
+      new GtfsCalendar.Builder()
```
Comment (KathleenQ) [inline gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidatorTest.java:38]: csvRowNumber should be included.

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java (hunk 2 above)
Comment (lilian-hunt) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java]: Good point! Every schema extends GtfsEntity which has csvRowNumber has the only attribute. So every table schema always stores the row number

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java (hunk 1 above)
Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: can only get 0 or 1

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java (hunk 2 above)
Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java]: yep I see. Will add this, thanks

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java (hunk 1 above)
Comment (jupi3200) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: But is this an implementation detail you should depend on?

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java (hunk 2 above)
Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java]: fixed

File: gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidatorTest.java (hunk 3 above)
Comment (aannie-y) [reply gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidatorTest.java:38]: yep fixed!

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java (hunk 2 above)
Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/notice/CalendarHasNoActiveDaysNotice.java]: Yep, i have removed all that and added a csv row number instead.

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java (hunk 1 above)
Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: the enum values for calendar days are only 0 (for not available) and 1 (for available).

Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: would just checking `calendar.monday().getNumber() === 'AVAILABLE'` for all the days be better then?

Comment (jupi3200) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: How about keeping a const variable for AVAILABLE number and then comparing each day with the specified number.

Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: Fixed. Added variable NOT_AVAILABLE = 0 and used that to compare with the days.getNumber() instead.

File: gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarTableContainer;
+
+/**
+ * Validates days in calendar.txt has at least one active (available) day.
+ *
+ * <p>Generated notices: {@link CalendarHasNoActiveDaysNotice}.
+ */
+@GtfsValidator
+public class CalendarHasNoActiveDaysValidator extends FileValidator {
+  private final static int NOT_AVAILBLE = 0;
```
Comment (jupi3200) [inline gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: Please don't trust current implementation details.

Comment (aannie-y) [reply gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/CalendarHasNoActiveDaysValidator.java]: Fixed. Changed 0 to GtfsServiceCalendar.NOT_AVAILABLE.

**Review by jupi3200 — APPROVED** [review]

**Pull request merged**